x-pack/solutions/observability/plugins/apm/public/components/shared/summary/build_url.ts

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import url from 'url';
+import type { Transaction } from '../../../../typings/es_schemas/ui/transaction';
+import type { Span } from '../../../../typings/es_schemas/ui/span';
+import type { APMError } from '../../../../typings/es_schemas/ui/apm_error';
+
+export const buildUrl = (item: Transaction | Span | APMError, urlFull: string | undefined) => {
+  if (urlFull) return urlFull;
+  // URL fields from Otel
+  const urlScheme = item?.url?.scheme;
+  const urlPath = item?.url?.path;
+  const serverAddress = item?.server?.address;
+  const serverPort = item?.server?.port;
+
+  const hasURLFromFields = urlScheme && serverAddress && serverPort;
+
+  return hasURLFromFields
+    ? url.format({
```

Comment:
Alternatively, you could use URL class. 

```ts
 new URL(urlPath ?? '', `${urlScheme}://${serverAddress}:${serverPort}`).toString()
```

Reply:
What is the concern with `url.format`? I saw it is widely used in apm and that's why I used it here 🤔

Reply:
It's been deprecated https://nodejs.org/api/url.html#urlformaturlobject. it's unlikely that it will be removed though

Reply:
Thank you for spotting that! With the `new URL`, there are some validations ([the port](https://developer.mozilla.org/en-US/docs/Web/API/URL/port#examples), for example )
So I wrapped it in a try / catch to make sure we don't throw an error in case of invalid / "dummy" data and log it instead.